app/components/admin/edit_settings_component.py

```diff
@@ -7,10 +7,45 @@ def edit_settings_component() -> str | Response:
     edit_settings_form = EditSettingsForm()
 
     if edit_settings_form.validate_on_submit():
-        return redirect(url_for("admin.settings"))
 
-    attributes = {
-        "edit_settings_form": edit_settings_form,
-    }
+        restaurant_name = edit_settings_form.restaurant_name.data
+        address = edit_settings_form.address.data
+        postal_code = edit_settings_form.postal_code.data
+        restaurant_description = edit_settings_form.restaurant_description.data
+        image = edit_settings_form.image.data
+
+        filename = None
+        if image:
```

Comment:
please use the `upload_file` function in `app/services/upload_image_service.py`. It is already implemented and secure to use.